File: 0455_AssignCookies/455.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution
{
public:
    int findContentChildren(vector<int> &g, vector<int> &s)
    {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int gs = g.size();
        int ss = s.size();

        int r = 0;

        int res = 0;
        while (res < gs && r < ss)
        {
            if (g[res] <= s[r])
            {
                res++;
            }

            r++;
        }

        return res;
    }
};

File: 0455_AssignCookies/455.swift
class Solution {
    func findContentChildren(_ g: [Int], _ s: [Int]) -> Int {
        var res: Int = 0
        let g = g.sorted()
        let s = s.sorted()
        var r = 0
       
        let gs = g.count
        let ss = s.count

        while res < gs && r < ss {
            defer { r += 1 }
            guard g[res] <= s[r] else { continue }
            res += 1
        }

        return res
    }
}

